astropy/visualization/mpl_normalize.py

```diff
@@ -226,3 +228,66 @@ def simple_norm(data, stretch='linear', power=1.0, asinh_a=0.1, min_cut=None,
     vmin, vmax = interval.get_limits(data)
 
     return ImageNormalize(vmin=vmin, vmax=vmax, stretch=stretch, clip=clip)
+
+
+# used in imshow_norm
+_norm_sig = inspect.signature(ImageNormalize)
+
+
+def imshow_norm(data, ax=None, imshow_only_kwargs={}, **kwargs):
+    """ A convenience function to call matplotlib's `matplotlib.pyplot.imshow`
+    function, using an `ImageNormalize` object as the normalization.
+
+    Parameters
+    ----------
+    data : 2D or 3D array-like - see `~matplotlib.pyplot.imshow`
+        The data to show. Can be whatever `~matplotlib.pyplot.imshow` and
+        `ImageNormalize` both accept.
+    ax : None or `~matplotlib.axes.Axes`
+        If None, use pyplot's imshow.  Otherwise, calls ``imshow`` method of the
+        supplied axes.
+    imshow_only_kwargs : dict
+        Arguments to be passed directly to `~matplotlib.pyplot.imshow` without
+        first trying `ImageNormalize`.  This is only for keywords that have the
+        same name in both `ImageNormalize` and `~matplotlib.pyplot.imshow` - if
+        you want to set the `~matplotlib.pyplot.imshow` keywords only, supply
+        them in this dictionary.
+
+    All other keyword arguments are parsed first by the `ImageNormalize`
+    initializer, then to`~matplotlib.pyplot.imshow`.
+
+    Notes
+    -----
+    The ``norm`` matplotlib keyword is not supported.
+    """
+    if 'X' in kwargs:
+        raise ValueError('Cannot give both ``X`` and ``data``')
+
+    if 'norm' in kwargs:
+        raise ValueError('There is no point in using imshow_norm if you give '
+                         'the ``norm`` keyword - use imshow directly if you '
+                         'want that.')
+
+    imshow_kwargs = dict(kwargs)
+
+    norm_kwargs = {'data': data}
+    for pname in _norm_sig.parameters:
+        if pname in kwargs:
+            norm_kwargs[pname] = imshow_kwargs.pop(pname)
+
+    for k, v in imshow_only_kwargs.items():
+        if k not in _norm_sig.parameters:
+            raise ValueError('Provided a keyword to imshow_only_kwargs ({}) '
+                             'that is not a keyword for ImageNormalize. This is'
```

Comment:
I don't understand this sentence - why can't I pass an argument in ``imshow_only_kwargs`` that's not an argument of ``ImageNormalize``? Is there actually any harm in doing that?

Reply:
It's not explicitly harmful except that it bypasses the "there should be one and preferably only one way to do it" principle.  That is, it will probably lead to different users of this function passing imshow keywords either via this dict *or* the function itself, and the intent of "convenience" is that you shouldn't have to use this argument at all.

Also, allowing that will require the code to make a decision about which overrides the other when you pass the same keyword in both here and in the function call.  Whereas in the current code there's no ambiguity there - any keyword in the function call that doesn't get consumed by `ImageNormalize` is the only-and-only possible source for `imshow`.  So it also helps us refuse the temptation to guess.

Reply:
Oh, but I did add a clarifying comment about this before doing the rebase just now.

Reply:
Ok that's fine, I don't feel strongly about this.